.github/workflows/scripts/start-cil-runner.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+# This script is used to start a CNCF CIL runner 
+
+token=$1
+hostname=$2
+
+# Generate random number from datastamp as the hostname of runner
+label=$(date +%N)
+
+hostname="$hostname-$label"
+echo "Creating CNCF CIL machine: $hostname..."
+
+# Use user_data_scripts to register the CNCF CIL runner as a self-hosted runner
+user_data_scripts="#cloud-config\nusers:\n    - default\n    - name: smp\n      groups: sudo, docker\n      sudo: ALL=(ALL) NOPASSWD:ALL\n      lock_passwd: true\nruncmd:\n    - [runuser, -l, smp, -c, \'mkdir actions-runner && cd actions-runner\']\n    - [runuser, -l, smp, -c, \'curl -o actions-runner-linux-x64-2.287.1.tar.gz -L https://github.com/actions/runner/releases/download/v2.287.1/actions-runner-linux-x64-2.287.1.tar.gz\']\n    - [runuser, -l, smp, -c, \'tar xzf ./actions-runner-linux-x64-2.287.1.tar.gz\']\n    - [runuser, -l, smp, -c, \'export RUNNER_ALLOW_RUNASROOT=1\']\n    - [runuser, -l, smp, -c, \'./config.sh --url https://github.com/$REPOSITORY --token $REG_TOKEN --labels $hostname >> github-action-registeration.log\']\n    - [runuser, -l, smp, -c, \'./run.sh >> github-action-registeration.log\']"
+
+# TODO: the options "operating_system", "facility", "plan" are hardcoded now, we should make them configurable
```

Comment:
Is this TODO still pending, would this also come in the scope of this PR or should we just make a new issue and document it there, till someone else picks it up?

Reply:
It should be in another PR instead of this one. This PR are the initial automation of self-hosted runner, so I won't include too many changes. And regarding this TODO, I think there is still something to discuss, it's a good idea to create an new issue and see if anyone can picks it up.

Reply:
Sure! Let's create another issue on this

Reply:
@gyohuangxin I'll let you take care of making an issue on this one as you would understand the details better.

Reply:
@hershd23 I created the issue #41 to track it.